dix/req.c

```diff
@@ -2077,8 +2140,13 @@ static void dix_rop_units_set(struct m0_dix_req *req)
 	 * machine lock to get consistent results.
 	 */
 	if (pm->pm_pver->pv_is_dirty &&
-	    !pool_failed_devs_tlist_is_empty(&pool->po_failed_devices))
+	    !pool_failed_devs_tlist_is_empty(&pool->po_failed_devices)) {
+		if (ENABLE_DTM0)
+			M0_IMPOSSIBLE("DTM0 can not operate when permanently"
```

Comment:
In this case, what can we do to handle the permanent failure?
Only stop operating and panic?

Reply:
Yes, since DTM0 does not assume permanent failures, it's a future plans to handle permanent and transient failures.